Mixpanel/SessionMetadata.m

```diff
@@ -0,0 +1,52 @@
+//
+//  SessionMetadata.m
+//  Mixpanel
+//
+//  Created by Yarden Eitan on 10/27/17.
+//  Copyright © 2017 Mixpanel. All rights reserved.
+//
+
+#import "SessionMetadata.h"
+
+@implementation SessionMetadata
+
+- (instancetype)initWithDispatchQueue:(dispatch_queue_t)queue {
+    self = [super init];
+    if (self) {
+        self.eventsCounter = 0;
+        self.peopleCounter = 0;
+        self.sessionID = 0;
+        self.sessionStartEpoch = 0;
+        self.trackingQueue = queue;
+    }
+    return self;
+}
+
+- (void)applicationDidBecomeActive {
```

Comment:
i think i'd prefer this to be a generic `reset` method or something and then `applicationDidBecomeActive` calls it and handles the `dispatch_async` stuff. that way this object is more just handling the state and the Mixpanel object handles queues and making sure stuff is happening in the right order